kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/EndOffsetsPoller.java

```diff
@@ -63,8 +64,9 @@ public class EndOffsetsPoller {
 
   public EndOffsetsPoller(
       final Map<String, ?> configs,
-      final ResponsiveMetrics metrics
-  ) {
+      final ResponsiveMetrics metrics,
+      final KafkaClientSupplier clientSupplier
```

Comment:
nit: can you just have this accept an admin client? I feel like KafkaClientSupplier should only be called by streams.